lib/ansible/modules/network/nxos/nxos_hsrp.py

```diff
@@ -239,6 +241,19 @@ def get_hsrp_group(group, interface, module):
     return hsrp
 
 
+def get_hsrp_group_unknown_enum(module, command, hsrp_table):
+    '''Some older NXOS images fail to set the attr values when using structured output and
+    instead set the values to <unknown enum>. This fallback method is a workaround that
+    uses an unstructured (text) request to query the device a second time.
+    'sh_preempt' is currently the only attr affected. Add checks for other attrs as needed.
+    '''
+    if 'unknown enum:' in hsrp_table['sh_preempt']:
+        cmd = {'output': 'text', 'command': command.split('|')[0]}
```

Comment:
Does the command fail with structured json output? Or is it just for a specific key?

If the command fails for json request:
We actually have the logic already implemented in module_utils https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/nxos/nxos.py#L164.
When `check_rc` is set to `retry_json`, run_commands will send the command the request with non structured format if json fails. 
You will just need to append the error string to if it is any different than `Invalid Command` or `Ambiguous Command` https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/nxos/nxos.py#L170

Reply:
Yes, this fails for nxapi as well.
The show command is successful with structured, it is only the `sh_preempt` key that is faulty.

```
[{u'TABLE_grp_detail': {u'ROW_grp_detail': {u'sh_active_router_addr': u'0.0.0.0',
 u'sh_group_num': u'100', u'sh_authentication_data': u'1234', u'sh_cur_hold_attr': u'sec',
 u'sh_ip_redund_name_attr': u'Default', u'sh_prio': u'25', u'sh_standby_router_addr': u'0.0.0.0',
 u'sh_vmac_attr': u'Default MAC', u'sh_cur_hello_attr': u'sec', u'sh_vip': u'192.0.2.2',
 u'sh_cur_hello': u'3', u'sh_vip_attr': u'config', u'sh_keystring_attr': u'unencrypted',
 u'sh_num_track_obj': u'0', u'sh_active_router_prio': u'0', u'sh_num_of_total_state_changes': u'0',
 u'sh_cfg_prio': u'25', u'sh_if_index': u'Ethernet1/1', u'sh_can_forward': u'unknown enum:<411125249>',
 u'sh_group_type': u'v4', u'sh_fwd_upper_threshold': u'25', u'sh_group_state': u'Initial',
 u'sh_cur_hold': u'10', u'sh_standby_router_prio': u'0', u'sh_fwd_lower_threshold': u'0', u'sh_group_version': u'v2',
 u'sh_preempt': u'unknown enum:<808728065>',   <-------------<< *****
 u'sh_vmac': u'0000.0c9f.f064', u'sh_ip_redund_name': u'hsrp-Eth1/1-100', 'sh_num_of_state_changes':
 u'0', u'sh_authentication_type': u'md5', u'sh_state_reason': u'(Interface Down)'}}}]
```